### PR TITLE
TCE-4255 feat(cicd): Integrate the CoPilot repo CI/CD Pipeline with TGCloud;

### PR DESCRIPTION
**Pull request by pengchengrun**

### CODE REVIEW

File: .github/workflows/pre-merge-test.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+name: Pre Merge Test
+
+on:
+  pull_request:
+    branches:
+      - dev
```
Comment (parkererickson-tg) [inline .github/workflows/pre-merge-test.yml:6]: Due to the branch naming, this should be `cloud-dev`

**Review by parkererickson-tg — CHANGES_REQUESTED** [review]

File: .github/workflows/pre-merge-test.yml (hunk 1 above)
Comment (billshitg) [reply .github/workflows/pre-merge-test.yml:6]: And this PR should be merged into the cloud-dev branch as well instead of dev

Comment (parkererickson-tg) [reply .github/workflows/pre-merge-test.yml:6]: To keep actions consistent, we should merge it into dev and cloud-dev

Comment (pengchengrun) [reply .github/workflows/pre-merge-test.yml]: i change branch name to cloud-dev

**Review by billshitg — APPROVED** [review]

**Review by parkererickson-tg — APPROVED** [review]

**Pull request merged**